pkg/broker/ingress/ingress_handler.go

```diff
@@ -73,32 +122,52 @@ func (h *Handler) getBroker(name, namespace string) (*eventingv1.Broker, error)
 	return broker, nil
 }
 
-func guessChannelAddress(name, namespace, domain string) string {
-	url := url.URL{
-		Scheme: "http",
-		Host:   fmt.Sprintf("%s-kne-trigger-kn-channel.%s.svc.%s", name, namespace, domain),
-		Path:   "/",
+func (h *Handler) guessChannelAddress(name, namespace, domain string) (*duckv1.Addressable, error) {
+	broker, err := h.getBroker(name, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	addr := &duckv1.Addressable{
+		URL: &apis.URL{
+			Scheme: "http",
+			Host:   fmt.Sprintf("%s-kne-trigger-kn-channel.%s.svc.%s", name, namespace, domain),
+			Path:   "/",
+		},
 	}
-	return url.String()
+	if broker.Status.Address != nil {
+		addr.CACerts = broker.Status.Address.CACerts
+		addr.URL.Scheme = broker.Status.Address.URL.Scheme
+	}
+	return addr, nil
 }
 
-func (h *Handler) getChannelAddress(name, namespace string) (string, error) {
+func (h *Handler) getChannelAddress(name, namespace string) (*duckv1.Addressable, error) {
 	broker, err := h.getBroker(name, namespace)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if broker.Status.Annotations == nil {
-		return "", fmt.Errorf("Broker status annotations uninitialized")
+		return nil, fmt.Errorf("broker status annotations uninitialized")
 	}
 	address, present := broker.Status.Annotations[eventing.BrokerChannelAddressStatusAnnotationKey]
 	if !present {
-		return "", fmt.Errorf("Channel address not found in broker status annotations")
+		return nil, fmt.Errorf("channel address not found in broker status annotations")
+	}
+
+	url, err := apis.ParseURL(address)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse channel address url")
 	}
-	return address, nil
-}
 
-func (h *Handler) Start(ctx context.Context) error {
-	return h.Receiver.StartListen(ctx, h)
+	addr := &duckv1.Addressable{
+		URL: url,
+	}
+	if broker.Status.Address != nil {
+		addr.CACerts = broker.Status.Address.CACerts
+		addr.URL.Scheme = broker.Status.Address.URL.Scheme
+	}
```

Comment:
There is an annotation with the CA certs, so I'd do something like this:
```suggestion
        var caCerts *string
        certs, present := broker.Status.Annotations[eventing.BrokerChannelCACertsStatusAnnotationKey]
	if present && certs != "" {
		caCerts = pointer.String(certs)
	}
	addr := &duckv1.Addressable{
		URL: url,
		CACerts: caCerts
	}
```